ocaml/xapi/xapi_host.ml

```diff
@@ -1366,45 +1366,92 @@ let certificate_sync ~__context ~host = Certificates.local_sync ()
 let get_server_certificate ~__context ~host =
   Certificates.get_server_certificate ()
 
+let add_certificate_to_db ~__context ~host certificate
+    root_fingerprints =
+  let date_of_ptime time = Date.of_float (Ptime.to_float_s time) in
+  let dates_of_ptimes (a, b) = (date_of_ptime a, date_of_ptime b) in
+  let not_before, not_after =
+    dates_of_ptimes (X509.Certificate.validity certificate)
+  in
+  let raw_fingerprint =
+    X509.Certificate.fingerprint Mirage_crypto.Hash.(`SHA256) certificate
+  in
+  let fingerprint = Certificates.pp_hash raw_fingerprint in
+  let uuid = Uuid.(to_string (make_uuid ())) in
+  let ref = Ref.make () in
+  let _type =
+    if List.mem raw_fingerprint root_fingerprints then `host_and_pool else `host
```

Comment:
Would `self_signed` be a better name than `host_and_pool`?

Reply:
I'm torn about this. It has a very specific meaning in the database:

```
         ; "host_and_pool", "Certificate that both identifies a host and is trusted by the whole pool."
```

Reply:
Let's leave it as it is. It's simpler if the name matches the name in the database